### DISCUSSION TITLE
Button documentation

### DISCUSSION THREAD

**Discussion by yuri-kiss** [TurboWarp/extensions#516] [Q&A] [answered]
Is there any button documentation, or no, I do understand there is a way to get the button to appear which I have done with: 
` {
          opcode: "emptyFunctionForButtons",
          blockType: "button",
          text: "TEST BUTTON",
        }`
I though it would run the "emptyFunctionForButtons" function, but that was not the case from my testing, Is there anyway to handle the button?

**Comment by LilyMakesThings** [comment, marked as answer]
as well as opcode, you need to define "func" 
` { opcode: "emptyFunctionForButtons", func: "emptyFunctionForButtons", blockType: "button", text: "TEST BUTTON", }`

**Reply by yuri-kiss** [reply]
fast I will try this!

**Reply by yuri-kiss** [reply]
it worked!, How did you know to do that?

**Reply by LilyMakesThings** [reply]
I've used buttons before they were added lol

Garbo also made an example extension for reference :)

**Reply by yuri-kiss** [reply]
where?

**Reply by LilyMakesThings** [reply]
https://github.com/TurboWarp/extensions/pull/469

**Reply by yuri-kiss** [reply]
thanks

**Comment by yuri-kiss** [comment]
[ANSWER](https://github.com/TurboWarp/extensions/discussions/516#discussioncomment-6080850)
